File: java/bob/src/main/java/Bob.java
/**
 * @author Cedric Zhuang
 */
public class Bob {
    String hey(String s) {
        String ret;
        if (s.trim().isEmpty()) {
            ret = "Fine. Be that way!";
        } else if (s.toUpperCase().equals(s) && !s.toLowerCase().equals(s)) {
            ret = "Whoa, chill out!";
        } else if (s.endsWith("?")) {
            ret = "Sure.";
        } else {
            ret = "Whatever.";
        }
        return ret;
    }
}

File: scala/bob/src/main/scala/Bob.scala
/**
  *
  * @author Cedric Zhuang
  *
  */
class Bob {
  def hey(q: String) = q.trim match {
    case s if s.isEmpty => "Fine. Be that way!"
    case s if s.toUpperCase == s && s.exists(_.isLetter) => "Whoa, chill out!"
    case s if s.endsWith("?") => "Sure."
    case _ => "Whatever."
  }
}
